### PR TITLE
Include milliseconds in DataDog timestamp format [Refs #152]

### PR DESCRIPTION
**Pull request by ladenedge**
### What does this PR do?

* Include milliseconds in the DataDog timestamp (ie. `@timestamp`)
* Add a unit test for both the new millisecond format and the OTel timestamp

### Motivation

My logs have no milliseconds!

image

### Notes

[Per StackOverflow](https://stackoverflow.com/a/65910279/222481), Go does not seem to have a millisecond format string built in.  And per #152, DataDog does not support nanoseconds.  So instead I've just included a manual format string with the usual millisecond precision for logs.

All tests pass for me.

```
C:\Source\opentelemetry-mapping-go\pkg\otlp\logs>go test
PASS
ok      github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/logs       0.419s
```

### CODE REVIEW

**Comment by ladenedge** [comment]
Sorry, didn't notice the changelog step.  Hopefully I did that correctly.  Thanks for your patience!

File: pkg/otlp/logs/logs_translator.go [hunk 1]
```diff
@@ -156,7 +155,7 @@ func Transform(lr plog.LogRecord, res pcommon.Resource, logger *zap.Logger) data
 	if lr.Timestamp() != 0 {
 		// we are retaining the nano second precision in this property
 		l.AdditionalProperties[otelTimestamp] = strconv.FormatInt(lr.Timestamp().AsTime().UnixNano(), 10)
-		l.AdditionalProperties[ddTimestamp] = lr.Timestamp().AsTime().Format(time.RFC3339)
+		l.AdditionalProperties[ddTimestamp] = lr.Timestamp().AsTime().Format("2006-01-02T15:04:05.000Z")
```
Comment (liustanley) [inline pkg/otlp/logs/logs_translator.go:158]: The previous value `time.RFC3339 = "2006-01-02T15:04:05Z07:00"` has the suffix of `07:00` in its format string. Is there any reason why this suffix is omitted here?

Comment (ladenedge) [reply pkg/otlp/logs/logs_translator.go]: Good question!  I have added the timezone format in to mirror `time.RFC3339`.

(And fixed the changelog, gah.)

**Review by liustanley — APPROVED** [review]
LGTM, thanks for submitting this PR! Feel free to merge it yourself and we'll include it in the next releases for the Datadog Agent and OTel Collector

**Comment by ladenedge** [comment]
> Feel free to merge it yourself and we'll include it in the next releases for the Datadog Agent and OTel Collector

Sorry, I must be misunderstanding.  Surely I don't have access to merge PRs?

image

What am I missing?

**Comment by liustanley** [comment]
> > Feel free to merge it yourself and we'll include it in the next releases for the Datadog Agent and OTel Collector
> 
> Sorry, I must be misunderstanding. Surely I don't have access to merge PRs?
> 
> image
> 
> What am I missing? (Or did you mean merge it in my local copy so I can make use of it immediately?)

Sorry I didn't realize you don't have permissions, I'll merge for you

**Pull request merged**